ReactAndroid/src/main/java/com/facebook/react/views/text/CustomTextTransformSpan.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.facebook.react.views.text;
+
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.text.style.ReplacementSpan;
+
+public class CustomTextTransformSpan extends ReplacementSpan {
+
+  /**
+   * A {@link ReplacementSpan} that allows declarative changing of text casing.
+   * CustomTextTransformSpan will change e.g. "foo" to "FOO", when passed UPPERCASE.
+   *
+   * This needs to be a Span in order to achieve correctly nested transforms
+   * (for Text nodes within Text nodes, each with separate needed transforms)
+   */
+
+  private final TextTransform mTransform;
+
+  public CustomTextTransformSpan(TextTransform transform) {
+    mTransform = transform;
+  }
+
+  @Override
+  public void draw(Canvas canvas, CharSequence text, int start, int end, float x, int top, int y, int bottom, Paint paint) {
+    CharSequence transformedText = transformText(text);
+    canvas.drawText(transformedText, start, end, x, y, paint);
+  }
+
+	@Override
+  public int getSize(Paint paint, CharSequence text, int start, int end, Paint.FontMetricsInt fm) {
+    CharSequence transformedText = transformText(text);
+    return Math.round(paint.measureText(transformedText, start, end));
+  }
+
+  private CharSequence transformText(CharSequence text) {
+    CharSequence transformed;
+
+    switch(mTransform) {
+      case UPPERCASE:
+        transformed = (CharSequence) text.toString().toUpperCase();
+        break;
+      case LOWERCASE:
+       transformed = (CharSequence) text.toString().toLowerCase();
+       break;
+      case CAPITALIZE:
+       transformed = (CharSequence) capitalize(text.toString());
+       break;
+      default:
+        transformed = text;
+    }
+
+    return transformed;
+  }
+
+  private String capitalize(String rawString) {
```

Comment:
This will need to be a bit more robust to match iOS's `[NSString capitalizedString]`. I did some testing and something like this seems to work well and produce the same result as iOS.

```java
import java.text.BreakIterator;

...

String text = ".aa\nbb\tcc  dd EE \r\nZZ I like to eat apples. 中文éé 我喜欢吃苹果。awdawd   ";

BreakIterator wordIterator = BreakIterator.getWordInstance();
wordIterator.setText(text);

StringBuilder res = new StringBuilder(text.length());
int start = wordIterator.first();
for (int end = wordIterator.next(); end != BreakIterator.DONE; end = wordIterator.next()) {
  String word = text.substring(start, end);
  if (Character.isLetterOrDigit(word.charAt(0))) {
    res.append(Character.toUpperCase(word.charAt(0)));
    res.append(word.substring(1).toLowerCase());
  } else {
    res.append(word);
  }
  start = end;
}
Log.d("test", res.toString());
```

Might be nice to also add some more complex test cases like this. Could be a good candidate for some unit tests too.

Reply:
Nice, thanks for this! I've used your suggestion in c580a505c1fb6fe75aa314ebbea28c328b59572a and included your test string in the test view, as well as writing some UTs.

Although, I'm struggling to run them and I've noticed this in the UTs BUCK file:

```
rn_robolectric_test(
    name = "views",
    # TODO Disabled temporarily until Yoga linking is fixed t14964130
    # srcs = glob(['**/*.java']),
    srcs = glob(["image/*.java"]),
```
I can't see that issue, or run the Text UTs without hitting the linking issue mentioned 🤔 (so my UTs... probably work? 😅)

Not sure if anyone knows a way around this?